extensions/amp-form/0.1/amp-form.js

```diff
@@ -422,6 +339,23 @@ function getUserValidityStateFor(element) {
 
 
 /**
+ * Updates class names on the element to reflect the active invalid types on it.
+ *
+ * TODO(#5005): Maybe propagate the invalid type classes to parents of the input as well.
+ *
+ * @param {!Element} element
+ */
+function updateInvalidTypesClasses(element) {
+  if (!element.validity) {
+    return;
+  }
+  for (const validationType in element.validity) {
```

Comment:
Does accessing `element.validity` cause it to validate? If so, we need to cache this into an attribute.